src/backend/x11/window_inner.rs

```diff
@@ -0,0 +1,238 @@
+/**
+A note for future contributors and maintainers:
+
+When editing this file, grab the nearest copy of the ICCCM. Following the ICCCM is paramount to
+X11 clients behaving properly and preventing scenarios such as windows not being resized in tiling
+window managers.
+
+Pay particular attention to "Section 4: Client to Window Manager Communication"
+
+A link to the ICCCM Section 4: https://tronche.com/gui/x/icccm/sec-4.html
+*/
+use crate::utils::{Logical, Size};
+
+use super::{Atoms, Window, WindowProperties, X11Error};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, Mutex, Weak,
+};
+use x11rb::{
+    connection::Connection,
+    protocol::{
+        xfixes::ConnectionExt,
+        xproto::{
+            self as x11, AtomEnum, ConnectionExt as _, CreateGCAux, CreateWindowAux, Depth, EventMask,
+            PropMode, Screen, UnmapNotifyEvent, WindowClass,
+        },
+    },
+    rust_connection::RustConnection,
+    wrapper::ConnectionExt as _,
+};
+
+impl From<Arc<WindowInner>> for Window {
+    fn from(inner: Arc<WindowInner>) -> Self {
+        Window(Arc::downgrade(&inner))
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct WindowInner {
+    // TODO: Consider future x11rb WindowWrapper
+    pub connection: Weak<RustConnection>,
+    pub id: x11::Window,
+    root: x11::Window,
+    pub atoms: Atoms,
+    pub size: Mutex<Size<u16, Logical>>,
+    /// Maintains the current cursor visibility state.
+    ///
+    /// Required since we cannot enable an already enabled cursor in xfixes.
+    cursor_visible: Arc<AtomicBool>,
```

Comment:
I wondered what this means and found https://www.x.org/releases/X11R7.7/doc/fixesproto/fixesproto.txt saying:

> 	If the client has made no outstanding HideCursor requests a BadMatch error is generated.

Uhm. Just ignore that error?
```rust
// This generates a Match error if we did not call HideCursor before. Ignore that error.
conn.xfixes_show_cursor().ignore_error();
```
That way this whole `cursor_visible` variable can go away, I think.